src/Items/Item.cs

```diff
@@ -27,5 +28,33 @@ public virtual bool IsLike(Item item)
         {
             return false;
         }
+
+		public bool Equals(Item other)
+		{
+			if (other.GetType () == GetType ())
+			{
+				foreach (var property in GetType().GetProperties())
+					if (!property.GetValue (other).Equals (property.GetValue (this)))
+						return false;
+				return true;
+			}
+			return false;
+		}
+
+		public override bool Equals (object obj)
+		{
+			if (obj is Item)
+				return Equals ((Item)obj);
+			return base.Equals (obj);
+		}
+
+		public override int GetHashCode ()
+		{
+			var elements = GetType ().GetProperties ().Select (p => p.GetValue (this).GetHashCode ());
+			int sum = 0;
+			foreach (var e in elements)
+				sum = (sum + e) % 288919320;

```

Comment:
Nie ma to większego znaczenia, ale na prezentacji o C# 7.1 na Buildzie pokazali jak VS będzie generować GetHashCode i tam właśnie robili modulo duża liczba ktora binarnie miała w miarę regularny układ binarny.